tests/18-warnp.sh

```diff
@@ -9,24 +9,34 @@ test_output="${s_basename}-stderr.txt"
 check_output() {
 	filename=$1
 	nonce=$2
+	is_syslog=$3
 
 	# Check that the output contains the expected nonce.
 	test "$(grep -c "nonce=${nonce}" "${filename}")" -eq "1"
 
 	# We should have 2 lines with "warnp errno".
 	test "$(grep -c "warnp errno" "${filename}")" -eq "2"
 
-	# We should have 7 lines in total.
-	test "$(wc -l < "${filename}")" -eq "7"
+	# Console or syslog?
+	if [ "${is_syslog}" -eq "0" ]; then
+		# Check console.
 
-	# We should have 2 lines without a colon.
-	test "$(grep -c -v "test_warnp:" "${filename}")" -eq "2"
+		# We should have 7 lines in total.
+		test "$(wc -l < "${filename}")" -eq "7"
```

Comment:
Why is this 2?  Is there a change to the test code which should be part of this commit?

Reply:
Oh, never mind, I got confused by how diff is matching up the lines.

Reply:
Yeah, I was annoyed by the way that diff was generated.